hypershift-operator/controllers/hostedcluster/internal/platform/aws/aws.go

```diff
@@ -43,7 +43,12 @@ func (p AWS) ReconcileCAPIInfraCR(hcluster *hyperv1.HostedCluster, controlPlaneN
 	if err != nil {
 		return nil, err
 	}
-
+	// reconciliation strips TypeMeta. We repopulate the static values since they are necessary for
+	// downstream reconciliation of the CAPI Cluster resource.
+	awsCluster.TypeMeta = metav1.TypeMeta{
```

Comment:
if this is being stripped how is this working atm though?

Reply:
it's not working for IBM Cloud. It only works for AWS because it happens to be at v1beta1